src/views/List.jsx

```diff
@@ -1,14 +1,48 @@
 import { ListItem } from '../components';
+import { useState, useEffect } from 'react';
 
 export function List({ data }) {
+	const [filterVal, setFilterVal] = useState('');
+	const [filteredList, setFilteredList] = useState([]);
+
+	const clearInput = (e) => {
+		e.preventDefault();
+		setFilterVal('');
+	};
+
+	useEffect(() => {
+		setFilteredList(data);
+	}, [data]);
+
+	useEffect(() => {
+		setFilteredList(
+			data.filter((item) =>
+				item.name.toLowerCase().includes(filterVal.toLowerCase()),
+			),
+		);
+	}, [filterVal]);
+
```

Comment:
**nitpick** **(non-blocking)**: Maybe combine two useEffect into one? This way, either `data` or `filterVal `is updated, it will trigger a re-render.

Reply:
**question:**  what does the first useEffect do? If I remove it and add `data` to the dependency array of the second one, it seems to work just fine.

Reply:
It was meant to give us access to the data prop at the top of the page render, since we were having issues rendering the filtered array. Though I guess we might not have tried to adding it to the dependent array in the second useEffect! Thanks for the tip, its working fine for me as well!